### ISSUE TITLE
build: allow for building of cmake projects which do not have package.xml

### ISSUE THREAD

**Issue by wjwwood** [catkin/catkin_tools#64]
The idea is that you can have a folder of plain CMake projects, which will all get built first, in parallel and in no particular order before trying to build any catkin packages. This allows you to easily build CMake projects as part of your workspace without creating package.xml's for them.

**Comment by NikolausDemmel** [comment]
> @dirk-thomas is right, when no --install is requested then the plain cmake packages are make install'ed into the devel space.

Is this actually supposed to still work? It seems for me it does not (i.e. the package is built but not installed into the devel space, using the latest binary release and tested with all 3 devel space options). I will do more testing and open another issue, but I was just wondering if people are using this feature. 

cc @mikepurvis, @wjwwood

**Comment by mikepurvis** [comment]
Yeah, it works. With catkin_tools 0.4.4, from a ros_base source workspace:

```
$ catkin build
------------------------------------------------------------------
Profile:                     default
Extending:                   None
Workspace:                   /Users/mikepurvis/ros_base_ws
------------------------------------------------------------------
Source Space:       [exists] /Users/mikepurvis/ros_base_ws/src
Log Space:         [missing] /Users/mikepurvis/ros_base_ws/logs
Build Space:        [exists] /Users/mikepurvis/ros_base_ws/build
Devel Space:        [exists] /Users/mikepurvis/ros_base_ws/devel
Install Space:      [unused] /Users/mikepurvis/ros_base_ws/install
DESTDIR:            [unused] None
------------------------------------------------------------------
Devel Space Layout:          merged
Install Space Layout:        None
------------------------------------------------------------------
Additional CMake Args:       None
Additional Make Args:        None
Additional catkin Make Args: None
Internal Make Job Server:    True
Cache Job Environments:      False
------------------------------------------------------------------
Whitelisted Packages:        None
Blacklisted Packages:        None
------------------------------------------------------------------
Workspace configuration appears valid.

NOTE: Forcing CMake to run for each package.
------------------------------------------------------------------
[build] Found '76' packages in 0.0 seconds.                                                                                                                  
[build] Updating package table.                                                                                                                              
Starting  >>> catkin                                                                                                                                         
Finished  <<< catkin                               [ 8.3 seconds ]                                                                                           
Starting  >>> genmsg                                                                                                                                         
Starting  >>> cmake_modules                                                                                                                                  
Starting  >>> cpp_common                                                                                                                                     
Starting  >>> rosbag_migration_rule                                                                                                                          
Finished  <<< rosbag_migration_rule                [ 6.0 seconds ]
^C
$ source devel/setup.bash
$ python -c "import catkin; print catkin.__file__"
/Users/mikepurvis/ros_base_ws/devel/lib/python2.7/site-packages/catkin/__init__.py
```

Noting that the catkin package itself is a `cmake` build: https://github.com/ros/catkin/blob/kinetic-devel/package.xml#L18-L19

Do you have a MWE of the issue you're seeing?

**Comment by NikolausDemmel** [comment]
Thanks for the datapoint Mike. I'll investigate and put together a MWE tomorrow.
